sdk/clientcore/core/src/main/java/io/clientcore/core/instrumentation/Instrumentation.java

```diff
@@ -127,15 +167,25 @@ public interface Instrumentation {
      *
      * @param applicationOptions Telemetry collection options provided by the application.
      * @param libraryOptions Library-specific telemetry collection options.
+     * @param serviceEndpoint The service endpoint.
      * @return The instance of telemetry provider implementation.
      */
     static Instrumentation create(InstrumentationOptions applicationOptions,
-        LibraryInstrumentationOptions libraryOptions) {
+        LibraryInstrumentationOptions libraryOptions, String serviceEndpoint) {
         Objects.requireNonNull(libraryOptions, "'libraryOptions' cannot be null");
+
+        String host = null;
+        int port = -1;
+        if (serviceEndpoint != null) {
+            URI uri = URI.create(serviceEndpoint);
```

Comment:
Do we want to accept `serviceEndpoint` as a `URI` instead of `String`?

Reply:
It's passed from the `ClientBuilder` and follows [`EndpointTrait.endpoint(String)`](https://github.com/Azure/azure-sdk-for-java/blob/a7b9c462584e712540018fe017676c650679f1ec/sdk/clientcore/core/src/main/java/io/clientcore/core/traits/EndpointTrait.java#L26), I changes it to string to avoid codegen doing conversion

Reply:
Makes sense.

@srnagar / @JonathanGiles, do we want to review `EndpointTrait` on whether it should be changed to `URI` rather than `String`?